docs/architecture/structure.md

```diff
@@ -0,0 +1,106 @@
+# Structure of the ETL
+
+## A compute graph
+
+The ETL is a compute graph, that uses a directed acyclic graph (DAG) to describe the dependencies between datasets. There are currently three types of nodes in the graph: snapshots, datasets and grapher exports.
+
+### Snapshots
+
+Data snapshots represent a copy of an upstream datasource in its original format, at a particular point in time.
+
+New-style snapshots are managed in the ETL using DVC and begin with the prefix `snapshot://`. Old-style snapshots are managed by the [walden](https://github.com/owid/walden) codebase and begin with the prefix `walden://`.
+
+Snapshot URIs use the convention: 
+
+```
+snapshot://<provider>/<version>/<filename>
+```
+
+Examples:
+
+- `snapshot://aviation_safety_network/2022-10-14/aviation_statistics.csv`
+- `walden://irena/2022-10-07/renewable_power_generation_costs`
+
+### Datasets
+
+Datasets are nodes in the compute graph, the main units of work in the ETL. They represent a transformation from one or more ingredients into a better, more useful output.
+
+Dataset URIs use the convention:
+
+```
+data://<channel>/<provider>/<version>/<dataset>
```

Comment:
We're mixing `provider` and `namespace`. Our code uses `namespace` so I'd stick to it and let them know it's typically provider.

Reply:
👍